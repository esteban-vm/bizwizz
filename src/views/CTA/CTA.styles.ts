import clsx from 'clsx'

const styles = {
  wrapper: clsx('flex w-full flex-col items-center justify-around bg-black p-12 md:flex-row'),
  links: clsx('flex flex-row'),
  title: clsx('mb-3 text-[4rem] font-semibold text-white md:mb-0'),
}

export default styles
