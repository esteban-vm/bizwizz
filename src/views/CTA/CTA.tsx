import type { FC } from '@/types'
import { Button, SocialLink, Wave } from '@/components'
import styles from './CTA.styles'

const CTA: FC = () => {
  return (
    <>
      <Wave color='white' />
      <section aria-labelledby='cta_title' className={styles.wrapper} id='call-to-action'>
        <div className={styles.links}>
          <SocialLink link='facebook' />
          <SocialLink link='twitter' />
          <SocialLink link='linkedin' />
        </div>
        <div className={styles.cta}>
          <h2 id='cta_title'>Let&apos;s Talk</h2>
        </div>
        <div className={styles.btn}>
          <Button />
        </div>
      </section>
    </>
  )
}

export default CTA
