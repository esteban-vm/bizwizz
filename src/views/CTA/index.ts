export { default as CTA } from './CTA'
