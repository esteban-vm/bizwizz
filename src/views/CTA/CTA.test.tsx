import { render, cleanup, within, screen } from '@/tests'
import CTA from './CTA'

describe('CTA:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<CTA />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /let's talk/i
    const ctaElem = screen.getByRole('region')
    expect(ctaElem).toBeInTheDocument()
    expect(ctaElem).toHaveAccessibleName(title)
    expect(ctaElem).toHaveAttribute('id', 'call-to-action')
    expect(screen.getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(ctaElem).getByRole('link', { description: /facebook/i })).toBeInTheDocument()
    expect(within(ctaElem).getByRole('link', { description: /twitter/i })).toBeInTheDocument()
    expect(within(ctaElem).getByRole('link', { description: /linkedin/i })).toBeInTheDocument()
    expect(within(ctaElem).getByRole('heading', { name: title, level: 2 })).toBeInTheDocument()
    expect(within(ctaElem).getByRole('button', { name: /get a quote today/i })).toBeInTheDocument()
  })
})
