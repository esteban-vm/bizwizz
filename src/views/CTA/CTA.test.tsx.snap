// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTA: > should render correctly 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="wave text-white"
    preserveAspectRatio="none"
    role="presentation"
    viewBox="0 0 500 150"
  >
    <path
      d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
      fill="currentColor"
      stroke="none"
    />
  </svg>
  <section
    aria-labelledby="cta_title"
    class="flex w-full flex-col items-center justify-around bg-black p-12 md:flex-row"
    id="call-to-action"
  >
    <div
      class="flex flex-row"
    >
      <a
        href="https://www.facebook.com"
        id="c10f7e60-6e36-498f-9a72-7b481133a14f"
        rel="noreferrer"
        target="_blank"
        title="facebook"
      >
        <svg
          class="social_link cta"
          role="img"
        >
          <use
            xlink:href="#icon-0"
          />
        </svg>
      </a>
      <a
        href="https://www.twitter.com"
        id="dc35c092-25ed-4754-8743-f4d9cd982004"
        rel="noreferrer"
        target="_blank"
        title="twitter"
      >
        <svg
          class="social_link cta"
          role="img"
        >
          <use
            xlink:href="#icon-2"
          />
        </svg>
      </a>
      <a
        href="https://www.linkedin.com"
        id="5780b1a7-ea08-4bf7-9076-685e21e4483a"
        rel="noreferrer"
        target="_blank"
        title="linkedin"
      >
        <svg
          class="social_link cta"
          role="img"
        >
          <use
            xlink:href="#icon-1"
          />
        </svg>
      </a>
    </div>
    <h2
      class="mb-3 text-[4rem] font-semibold text-white md:mb-0"
      id="cta_title"
    >
      Let's Talk
    </h2>
    <button
      class="button"
      type="button"
    >
      Get a quote today
    </button>
  </section>
</DocumentFragment>
`;
